agent/mapcall/src/main/java/randoop/instrument/ReplacementFileReader.java

```diff
@@ -0,0 +1,524 @@
+
+  /**
+   * Naive regex to match a method signature as a sequence of period-delimited identifiers followed
+   * by a string enclosed in parentheses. Does not actually check that the form of the arguments is
+   * correct.
+   */
+  private static final String SIGNATURE_STRING = PACKAGE_OR_CLASS_STRING + "\\([^)]*\\)";
+
+  /**
+   * Naive regex to match a method signature with groups to extract the fully-qualified method name
+   * (group 1) and type argument list (group 3). (Group 2 is the group from {@link
+   * #PACKAGE_OR_CLASS_STRING}.)
+   */
+  private static final String SIGNATURE_GROUP_STRING =
+      "(" + PACKAGE_OR_CLASS_STRING + ")" + "\\(([^)]*)\\)";
+
+  /** Pattern to match naive signature regex with groups. */
+  private static final Pattern SIGNATURE_PATTERN = Pattern.compile(SIGNATURE_GROUP_STRING);
+
+  /**
+   * Pattern to match of method replacements consisting of a pair of signatures. Uses the naive
+   * signature pattern separated by one or more spaces or tabs. Can be used to decide whether a
+   * replacement file line indicates a method replacement. Groups 1 and 3 correspond to each of the
+   * signature strings.
+   */
+  private static final Pattern SIGNATURE_LINE =
+      Pattern.compile("(" + SIGNATURE_STRING + ")[ \\t]+(" + SIGNATURE_STRING + ")");
+
+  /**
+   * Pattern to match class or package replacements consisting of a pair of class or package name
+   * signatures. Groups 1 and 3 correspond to each of the package or class names.
+   */
+  private static final Pattern PACKAGE_OR_CLASS_LINE =
+      Pattern.compile(
+          "(" + PACKAGE_OR_CLASS_STRING + ")[ \\t]+" + "(" + PACKAGE_OR_CLASS_STRING + ")");
+
+  /**
+   * Reads the given replacement file specifying method calls that should be replaced by other
+   * method calls. See the <a
+   * href="https://randoop.github.io/randoop/manual/index.html#map_calls">mapcall user
+   * documentation</a> for details on the file format.
+   *
+   * @param map_file the file with map of method substitutions
+   * @throws IOException if there is an error reading the file
+   */
+  static ConcurrentHashMap<MethodDef, MethodDef> readFile(File map_file)
+      throws IOException, ReplacementFileException {
+    return readFile(new FileReader(map_file));
+  }
+
+  /**
+   * Reads the replacement file specifying method calls that should be replaced by other method
+   * calls. See the <a href="https://randoop.github.io/randoop/manual/index.html#map_calls">mapcall
+   * user documentation</a> for details on the file format.
+   *
+   * @param in the {@code Reader} for the replacement file
+   * @throws IOException if there is an error reading from the file
+   */
+  static ConcurrentHashMap<MethodDef, MethodDef> readFile(Reader in)
+      throws IOException, ReplacementFileException {
+    ConcurrentHashMap<MethodDef, MethodDef> replacementMap = new ConcurrentHashMap<>();
+    LineNumberReader lr = new LineNumberReader(in);
```

Comment:
MapCallsAgent uses
```    try (EntryReader reader = new EntryReader(exclusionReader, filename, "//.*$", null)) {```
Is there a reason that this code does not?
(It's a plume-lib API, so a bit non-standard, but is shorter to read, and you were willing to use it elsewhere.)